test/acl_profiler_test.cpp

```diff
@@ -611,7 +611,9 @@ MT_TEST_GROUP(acl_no_profile) {
   enum { MAX_DEVICES = 100, m_num_devices_in_context = 3 };
   void setup() {
     if (threadNum() == 0) {
-      remove(PROFILE_MON);
+      // Remove PROFILE_MON if it exists. We don't care if this call passes or
+      // not since we just want to guarantee that PROFILE_MON is gone.
+      (void)remove(PROFILE_MON);
```

Comment:
This doesn't work. The PROFILE_MON file may not exist to begin with. I should have explained this in the comment.

Alternatively, we could add a check to see if the file exists, and then implement it like you said.

Reply:
> This does not apply here, but it's a good opportunity to discuss how this would be solved in program (not test) code. In general, when a file operation depends on the existence of the file, checking whether it exists separately introduces a race condition; even if the file existed at the time of the check, it might still have been deleted between the check and the subsequent file operation through other means since the file system is a shared resource.
> 
> The safe and diligent way is to do the file operation, e.g., `remove()` without checking for existence, then check the error code, [`errno` ](https://man7.org/linux/man-pages/man3/errno.3.html) for `ENOENT` which indicates that the path does not exist; otherwise return the error.

Aha! I haven't thought about that. This makes a lot of sense. Thanks for the insight!